core/src/ecs/systems/core/PhysicsSystem.ts

```diff
@@ -104,22 +104,28 @@ export const PhysicsSystem: SystemBuilder<"PhysicsSystem"> = {
         });
 
         // sensor callbacks
-        values(colliders).forEach((collider: Collider) => {
+        for (const [entity, collider] of colliders.entries()) {
           if (collider.sensor && collider.rapierCollider) {
+            
             const collidedWith: Entity<Collider | Position>[] = [];
 
             physics.intersectionPairsWith(collider.rapierCollider, (collider2) => {
-              const collided = entries(colliders).find(([_, c]) => c.rapierCollider === collider2);
-              if (collided && world.entities[collided[0]]) collidedWith.push(world.entities[collided[0]] as Entity<Collider | Position>);
+              const collided = colliders.entries().find(([_, c]) => c.rapierCollider === collider2);
+              if (collided && world.entities[collided[0].id]) collidedWith.push(world.entities[collided[0].id] as Entity<Collider | Position>);
             });
 
             // collide only once
             let collided = false;
-            collidedWith.sort((a, b) => b.components.collider.priority - a.components.collider.priority).forEach((entity) => {
+
+            collidedWith.sort((a, b) => {
+              const aDistance = XYdistance(a.components.position.data, entity.components.position.data);
+              const bDistance = XYdistance(b.components.position.data, entity.components.position.data);
+              return aDistance > bDistance ? 1 : -1;
+            }).forEach((entity) => {
```

Comment:
refactored `colliders` here and the sorting logic to sort by distance

but the distance here should be the "source" entity, not the hitbox entity, so it's still not perfect

may be better to just change the hitbox instead of this sorting logic, not sure yet. something like — let the sword hit everyone in the hitbox, or make it a growing hitbox